### DISCUSSION TITLE
Add to right click context menu? (Windows)

### DISCUSSION THREAD

**Discussion by Andersama** [helix-editor/helix#8081] [General]
I just looked up partly how to do this, it's a fairly convenient feature I've found for things like git gui and git bash and other programs as I'm navigating folders. There are two keys to add to the registry, one works when specifically clicking on a folder item the other on the background. I'm not sure the key to add for handling files* (that'd also be useful) but here's a start.

```
HKEY_CLASSES_ROOT\Directory\Background\shell\hx (Default) REG_SZ hx Here
HKEY_CLASSES_ROOT\Directory\Background\shell\hx Icon <path_to_hx>
HKEY_CLASSES_ROOT\Directory\Background\shell\hx\command "<path_to_hx>" "%V"

HKEY_CLASSES_ROOT\Directory\shell\hx (Default) REG_SZ hx Here
HKEY_CLASSES_ROOT\Directory\shell\hx Icon <path_to_hx>
HKEY_CLASSES_ROOT\Directory\shell\hx\command "<path_to_hx>" "%1"
```

**Comment by pascalkuthe** [comment]
I don't really see that as something that should be supported by helix. We do not ship any kind of windows installer (which would be needed to create such keys) and don't intend to. Helix is also a tui application so the intended usecase it to launch from the terminal (not windows explorer). You are free to explore this on your own ofcourse or maybe even try to upstream it into something like scoop but those are not maintained by us so this does not need to be tracked in this repo.

**Reply by Andersama** [reply]
So far as I'm aware, I think programs can attempt to modify the registry themselves although I'm sure it takes admin privileges, I don't think it requires an installer. I tend to launch my terminal from the folder explorer so having an option to do the same but with helix is nice, and of course since it's a terminal application provided the context menu passes the folder or files paths over it can be a faster way to jump straight to editing a specific file.

**Reply by Andersama** [reply]
Also as nice as being able to install from source with a few commands, it's probably worth having an installer, compiling from source on my end takes quite a while, my computer may be old, but it's not a slouch. There's some tools out there I'm sure to automate creating an installer per build with CI I deal with c++ though I'm not sure if there's something similar for rust.

**Reply by CptPotato** [reply]
I personally just use a very basic launcher program on Windows that opens a file with helix in my terminal emulator. With that I can just set up the file associations and open files like that, without having to fiddle around with the registry manually.

> Also as nice as being able to install from source with a few commands, it's probably worth having an installer, compiling from source on my end takes quite a while

You can already get helix through a variety of package managers available for the common operating systems. Even Windows comes with winget pre-installed these days it seems, so getting helix is a matter of a few minutes at most.

**Comment by lMINERl** [comment]
Although there is nothing to add here , I was looking for a way to open helix current file with current workspace directory in a wezterm in a right click context menu and this discussion pointed me to right direction 
so I come up with this ( just to help people who is looking for the same thing )

some references : 
[https://www.robvanderwoude.com/ntstart.php](url)
[https://scoop.sh/#/apps?q=helix+%22https%3A%2F%2Fgithub.com%2FScoopInstaller%2FMain%22](url)

`powershell -WindowStyle hidden -Command "(Set-Location %w) ; (<path_to_wezterm>\wezterm-gui.exe --config-file <path_to_config>\.wezterm.lua start --cwd %w -- hx  --working-dir %w '%1' )"`
